msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CodesignTaskBase.cs

```diff
@@ -110,11 +164,11 @@ public override bool Execute ()
 			if (Resources.Length == 0)
 				return true;
 
-			EnvironmentVariables = new string[] {
-				"CODESIGN_ALLOCATE=" + CodesignAllocate
-			};
+			Parallel.ForEach (Resources, new ParallelOptions { MaxDegreeOfParallelism = 4 }, (item) => {
```

Comment:
Just did my own test and it does look like Xcode uses `--timestamp=none` for device builds (Product -> Build). They *also* seem to use `--timestamp=none` for "Product -> Archive" builds (and yes, they rebuild when Archiving which was not what I expected).